management/hostedservice/client.go

```diff
@@ -117,6 +117,19 @@ func (self HostedServiceClient) GetHostedService(name string) (HostedService, er
 	return hostedService, nil
 }
 
+func (self HostedServiceClient) ListHostedServices() ([]HostedService, error) {
+	var response listHostedServiceResponse
+
+	data, err := self.client.SendAzureGetRequest(azureHostedServiceListURL)
+	if err != nil {
+		return response.HostedServices, err
+	}
+
+	err = xml.Unmarshal(data, &response)
+
+	return response.HostedServices, err
```

Comment:
@paulmey ideally we want to return `ListHostedServiceResponse` which could be a typedef for `[]HostedService`. 

What they do in c# in, `class ListHostedServiceResponse : IList<HostedService>` however the interface is implemented through a field on `response.HostedServices` like you have in here.

I'm trying to see if we can avoid such internal fields + if tomorrow ListHostedServices returns a response along with the list (I don't know but not hard to imagine this happening) then the signature will need to change and you'll break people.

That's why we probably want to return `ListHostedServiceResponse` type here (and in many other places) :smile_cat: